### PR TITLE
stationarity + cointegration + regression

### PR DESCRIPTION
**Pull request by akwon31**
# Description

Beginning of a new statistical arbitrage module.

The module will be separated into multiple parts, and I will be making PR's for each submodule.

Some docs have a placeholder of `To be implemented`.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Operating system: PyCharm
* IDE used: Mac

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/statistical_arbitrage/stationarity.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    The p-values are obtained through regression surface approximation from
+    MacKinnon 1994, but using the updated 2010 tables. If the p-value is close
+    to significant, then the critical values should be used to judge whether
+    to reject the null.
+
+    :param data: (pd.Series) The data series to test.
+
+    :param maxlag: (int) Maximum lag which is included in test, default :math:`\frac{12 * nobs}{100}^{1/4}`.
+
```
[the remark this answers is not in the document]
Comment (akwon31) [reply mlfinlab/statistical_arbitrage/stationarity.py:36]: Do you know why it's not working? Can't seem to figure out why it doesn't render on sphinx.

[the remark this answers is not in the document]
Comment (akwon31) [reply mlfinlab/statistical_arbitrage/stationarity.py:36]: https://stackoverflow.com/questions/16468397/mathjax-expression-in-sphinx-python-not-rendering-correclty%22

Fixed it! If you put inline math equations in your python docstrings you have to double backslash everything

File: mlfinlab/statistical_arbitrage/stationarity.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    :param regression: (str) (``c``, ``ct``, ``ctt``, ``nc``) Constant and trend order to include in
+        regression.
+
+        - ``c`` : constant only (default).
```
[the remark this answers is not in the document]
Comment (akwon31) [reply mlfinlab/statistical_arbitrage/stationarity.py:40]: https://github.com/sphinx-doc/sphinx/issues/4220

Apparently it is a known bug that they haven't fixed yet.

[the remark this answers is not in the document]
Comment (PanPip) [reply mlfinlab/statistical_arbitrage/stationarity.py]: Oh. Let's hope they will fix it soon 😅

[the remark this answers is not in the document]
Comment (mauro835) [reply mlfinlab/statistical_arbitrage/stationarity.py]: Do you know if this bug has been fixed?

[the remark this answers is not in the document]
Comment (PanPip) [reply mlfinlab/statistical_arbitrage/stationarity.py]: @mauro835, Just checked. This issue still exists in sphinx-3.2.1

**Pull request merged**